onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -147,23 +206,47 @@ Status EmbedLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
         word_gather_node.MutableInputDefs()[0],
         position_gather_node.MutableInputDefs()[0],
         segment_gather_node.MutableInputDefs()[0],
-        skip_ln_node.MutableInputDefs()[2],
-        skip_ln_node.MutableInputDefs()[3]};
+        layer_norm_node.MutableInputDefs()[1],
+        layer_norm_node.MutableInputDefs()[2]};
     Node& embed_layer_norm_node = graph.AddNode(graph.GenerateNodeName("EmbedLayerNormalization"),
-                                                "EmbedLayerNormalization", 
-                                                "fused EmbedLayerNorm subgraphs ", 
+                                                "EmbedLayerNormalization",
+                                                "fused EmbedLayerNorm subgraphs ",
                                                 embed_layer_norm_input_defs,
-                                                {skip_ln_node.MutableOutputDefs()[0], reduce_sum_node.MutableOutputDefs()[0]},
+                                                {layer_norm_node.MutableOutputDefs()[0], reduce_sum_node.MutableOutputDefs()[0]},
                                                 {}, kMSDomain);
 
     // Assign provider to this new node. Provider should be same as the provider for old node.
-    embed_layer_norm_node.SetExecutionProviderType(skip_ln_node.GetExecutionProviderType());
+    embed_layer_norm_node.SetExecutionProviderType(layer_norm_node.GetExecutionProviderType());
 
     // move input edges to gather (first in list) across to the embed_layer_norm_node.
     // move output definitions and output edges to embed_layer_norm_node.
     // remove all the other nodes.
     std::vector<NodeIndex> nodes_to_remove;
     if (p_shape_node != nullptr && p_expand_node != nullptr) {
+      // Match Shape --> Gather --> Unsqueeze --> ConstantOfShape --> NonZero --> Transpose --> Squeeze --> Cast --> Unsqueeze --> Expand
+      if (p_expand_node != nullptr) {
+        Node& expand_node = *graph.GetNode(p_expand_node->Index());
+        std::vector<graph_utils::EdgeEndToMatch> expand_parent_path{
+            {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
+            {0, 0, "Cast", {9}, kOnnxDomain},
+            {0, 0, "Squeeze", {1}, kOnnxDomain},
+            {0, 0, "Transpose", {1}, kOnnxDomain},
+            {0, 0, "NonZero", {9}, kOnnxDomain},
+            {0, 0, "ConstantOfShape", {9}, kOnnxDomain},
+            {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
+            {0, 0, "Gather", {1, 11}, kOnnxDomain},
+            {0, 0, "Shape", {1}, kOnnxDomain},
+        };
```

Comment:
Also check that shape's parent is input_ids. The edges of * has not been checked in your code:

input_ids -*- Shape -*- Expand -- Gather
   |*                                  |
   --Shape--(this Path) --